### PR TITLE
Various fixes in api.php

### PR DESCRIPTION
**Pull request by Rudloff**
I ran some code analysis tools in order to check my API code and also noticed some easy to fix errors in other people's code, so this PR regroups these various fixes.

I separated the commits in order to make the changes easier to read, but I can squash them if your prefer.

### CODE REVIEW

**Comment by MrPetovan** [comment]
What tool did you use? I’ve used PHPStan but it apparently can only check code in the `src` folder.

File: include/api.php [hunk 1]
```diff
@@ -3365,17 +3363,6 @@ function api_ff_ids($type,$qtype)
 
 	$user_info = api_get_user($a);
 
-	if ($qtype == 'friends') {
```
Comment (MrPetovan) [inline include/api.php]: Why did you remove this code?

Comment (MrPetovan) [reply include/api.php]: Ha, saw the commit log, never mind.

**Comment by Rudloff** [comment]
I used [Scrutinizer](https://scrutinizer-ci.com/) which is CI tool that runs a lot of different checks.
(You can have a look at the very verbose report [here](https://scrutinizer-ci.com/g/Rudloff/friendica/issues/develop).)

File: include/api.php (hunk 1 above)
Comment (Rudloff) [reply include/api.php]: The `$sql_extra` variable is not used anywhere in the function.

**Comment by MrPetovan** [comment]
Thanks, I’m asking because @annando tasked me with enabling a static code analysis check before each PR is merged, and so far I’ve considered and discarded both psalm (target PHP version is 7+) and PHPStan (doesn’t appear to adequately cover include-style projects like ours).

Of course we would need to do a huge first pass before wecan actually enforce any static code analysis check but choosing the right tool for this job is the furst step in the whole process.

**Comment by annando** [comment]
It happened too often in the past that people submitted badly tested code (including myself). So this would be a great first step.
